presto-main/src/main/java/com/facebook/presto/sql/planner/planPrinter/PlanPrinter.java

```diff
@@ -1223,10 +1224,17 @@ private String formatDomain(Domain domain)
 
         private void printCost(int indent, PlanNode... nodes)
         {
-            String costString = Joiner.on("/").join(Arrays.stream(nodes)
-                    .map(this::formatCost)
-                    .collect(toImmutableList()));
-            print(indent, "Cost: %s", costString);
+            if (Arrays.stream(nodes).anyMatch(this::isKnownCost)) {
+                String costString = Joiner.on("/").join(Arrays.stream(nodes)
+                        .map(this::formatCost)
+                        .collect(toImmutableList()));
+                print(indent, "Cost: %s", costString);
+            }
+        }
+
+        private boolean isKnownCost(PlanNode node)
+        {
+            return !Objects.equals(UNKNOWN_COST, costs.getOrDefault(node.getId(), UNKNOWN_COST));
```

Comment:
Since this is comparing with a constant object, no need to use `Objects.equals`
```java
return !UNKNOWN_COST.equals(costs.getOrDefault(node.getId(), UNKNOWN_COST));